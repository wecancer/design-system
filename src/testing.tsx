import React from 'react'
import {render, RenderResult} from '@testing-library/react'
import WecancerProvider from './WecancerProvider'

type Props = {
  children: React.ReactNode
}

const AllTheProviders = ({children}: Props) => {
  return <WecancerProvider>{children}</WecancerProvider>
}

// eslint-disable-next-line
const appRender = (ui: React.ReactElement, options?: any): RenderResult =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react'

// override render method
export {appRender as render}
