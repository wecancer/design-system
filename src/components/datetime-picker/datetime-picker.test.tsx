import DateTimePicker from '.'
import { render, fireEvent } from '../../testing'

describe('<DateTimePicker />', () => {
  it('should render snapshot correctly', () => {
    const { container } = render(
      <>
        <DateTimePicker
          datetime={new Date(0, 0, 0, 0, 0)}
          onChange={() => null}
        />
      </>,
    )
    fireEvent.click(container)
    expect(container).toMatchSnapshot()
  })
})
