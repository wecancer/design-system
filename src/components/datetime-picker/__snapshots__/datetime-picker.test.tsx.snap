// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateTimePicker /> should render snapshot correctly 1`] = `
<div>
  <div
    class="sc-bqiRlB gMMNus"
  >
    <div
      class="sc-furwcr gLQxJk"
      role="button"
      tabindex="0"
    >
      <div
        class="sc-gsDKAQ kdpXcC"
      >
        <input
          class="sc-dkPtRN kNILkx"
          placeholder="24/05/2022"
          type="text"
          value="31/12/1899"
        />
        <div
          aria-label="dd/mm/aaaa"
          class="sc-gKclnd fTVuoW"
          data-label="true"
          role="button"
          tabindex="-1"
        >
          dd/mm/aaaa
        </div>
        <button
          class="sc-eCImPb kfvWEQ"
          type="button"
        >
          <div
            class="sc-bdvvtL bYGNWJ icon-right"
            data-icon-name="calendar"
          >
            <svg
              fill="none"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="19"
                rx="3"
                stroke="currentColor"
                stroke-width="2"
                width="20"
                x="2"
                y="3"
              />
              <path
                d="M7 1V3"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M17 1V3"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M2 8H22"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M6.5 13H7.5"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M11.5 13H12.5"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M16.5 13H17.5"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M6.5 17H7.5"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M11.5 17H12.5"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M16.5 17H17.5"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
        </button>
        
      </div>
    </div>
    <div
      class="sc-pVTFL gOVlmP"
    >
      <div
        class="sc-jrQzAO jBebem"
      >
        <input
          class="sc-iqseJM gOYKcQ"
          max="23"
          min="0"
          placeholder="00"
          type="number"
          value="0"
        />
        :
        <input
          class="sc-iqseJM gOYKcQ"
          max="59"
          min="0"
          placeholder="00"
          type="number"
          value="0"
        />
      </div>
      <div
        class="sc-kDTinF PKwbv"
      >
        <div
          class="sc-crHmcD cqitUV"
        >
          <ul>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="00"
                type="button"
                value="00"
              >
                00
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="01"
                type="button"
                value="01"
              >
                01
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="02"
                type="button"
                value="02"
              >
                02
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="03"
                type="button"
                value="03"
              >
                03
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="04"
                type="button"
                value="04"
              >
                04
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="05"
                type="button"
                value="05"
              >
                05
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="06"
                type="button"
                value="06"
              >
                06
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="07"
                type="button"
                value="07"
              >
                07
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="08"
                type="button"
                value="08"
              >
                08
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="09"
                type="button"
                value="09"
              >
                09
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="10"
                type="button"
                value="10"
              >
                10
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="11"
                type="button"
                value="11"
              >
                11
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="12"
                type="button"
                value="12"
              >
                12
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="13"
                type="button"
                value="13"
              >
                13
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="14"
                type="button"
                value="14"
              >
                14
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="15"
                type="button"
                value="15"
              >
                15
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="16"
                type="button"
                value="16"
              >
                16
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="17"
                type="button"
                value="17"
              >
                17
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="18"
                type="button"
                value="18"
              >
                18
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="19"
                type="button"
                value="19"
              >
                19
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="20"
                type="button"
                value="20"
              >
                20
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="21"
                type="button"
                value="21"
              >
                21
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="22"
                type="button"
                value="22"
              >
                22
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="23"
                type="button"
                value="23"
              >
                23
              </button>
            </li>
          </ul>
        </div>
        <div
          class="sc-crHmcD cqitUV"
        >
          <ul>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="00"
                type="button"
                value="00"
              >
                00
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="01"
                type="button"
                value="01"
              >
                01
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="02"
                type="button"
                value="02"
              >
                02
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="03"
                type="button"
                value="03"
              >
                03
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="04"
                type="button"
                value="04"
              >
                04
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="05"
                type="button"
                value="05"
              >
                05
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="06"
                type="button"
                value="06"
              >
                06
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="07"
                type="button"
                value="07"
              >
                07
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="08"
                type="button"
                value="08"
              >
                08
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="09"
                type="button"
                value="09"
              >
                09
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="10"
                type="button"
                value="10"
              >
                10
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="11"
                type="button"
                value="11"
              >
                11
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="12"
                type="button"
                value="12"
              >
                12
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="13"
                type="button"
                value="13"
              >
                13
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="14"
                type="button"
                value="14"
              >
                14
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="15"
                type="button"
                value="15"
              >
                15
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="16"
                type="button"
                value="16"
              >
                16
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="17"
                type="button"
                value="17"
              >
                17
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="18"
                type="button"
                value="18"
              >
                18
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="19"
                type="button"
                value="19"
              >
                19
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="20"
                type="button"
                value="20"
              >
                20
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="21"
                type="button"
                value="21"
              >
                21
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="22"
                type="button"
                value="22"
              >
                22
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="23"
                type="button"
                value="23"
              >
                23
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="24"
                type="button"
                value="24"
              >
                24
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="25"
                type="button"
                value="25"
              >
                25
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="26"
                type="button"
                value="26"
              >
                26
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="27"
                type="button"
                value="27"
              >
                27
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="28"
                type="button"
                value="28"
              >
                28
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="29"
                type="button"
                value="29"
              >
                29
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="30"
                type="button"
                value="30"
              >
                30
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="31"
                type="button"
                value="31"
              >
                31
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="32"
                type="button"
                value="32"
              >
                32
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="33"
                type="button"
                value="33"
              >
                33
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="34"
                type="button"
                value="34"
              >
                34
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="35"
                type="button"
                value="35"
              >
                35
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="36"
                type="button"
                value="36"
              >
                36
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="37"
                type="button"
                value="37"
              >
                37
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="38"
                type="button"
                value="38"
              >
                38
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="39"
                type="button"
                value="39"
              >
                39
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="40"
                type="button"
                value="40"
              >
                40
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="41"
                type="button"
                value="41"
              >
                41
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="42"
                type="button"
                value="42"
              >
                42
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="43"
                type="button"
                value="43"
              >
                43
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="44"
                type="button"
                value="44"
              >
                44
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="45"
                type="button"
                value="45"
              >
                45
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="46"
                type="button"
                value="46"
              >
                46
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="47"
                type="button"
                value="47"
              >
                47
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="48"
                type="button"
                value="48"
              >
                48
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="49"
                type="button"
                value="49"
              >
                49
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="50"
                type="button"
                value="50"
              >
                50
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="51"
                type="button"
                value="51"
              >
                51
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="52"
                type="button"
                value="52"
              >
                52
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="53"
                type="button"
                value="53"
              >
                53
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="54"
                type="button"
                value="54"
              >
                54
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="55"
                type="button"
                value="55"
              >
                55
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="56"
                type="button"
                value="56"
              >
                56
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="57"
                type="button"
                value="57"
              >
                57
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="58"
                type="button"
                value="58"
              >
                58
              </button>
            </li>
            <li>
              <button
                class="sc-egiyK dfaVKW"
                id="59"
                type="button"
                value="59"
              >
                59
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
